### ISSUE TITLE
Download Model Stuck

### ISSUE THREAD

**Issue by sachinshelke** [cloud-annotations/object-detection-react#15]
Hi,
Model size is around 22 MB. many time is download smoothly but when slow internet detected, downloading stuck at 92%. 
here is the implementation

```
await tf.loadGraphModel( MODEL_PATH, 
       {
           onProgress: (progress)=> {
                   console.log('Model Progress' + progress)
           }
      }
)
```
Have tried to call this function again but still no it stuck at this at 92% and does not further proceed.

**How to reproduce?**
Clear cache,
in inspect mode --> Network --> Network Speed --> Slow 3G

After couple of iteration of above step it can be  reproduce.

_Understand that this is not related to ml, but still asking if get any solution._

**Comment by bourdakos1** [comment]
Just to clarify, does the model never load? or do you just never get a progress callback for 100%?

**Comment by sachinshelke** [comment]
Its 7:3 ratio. Many time it loads and get the callback of 100% as well however may time it stuck at 92%.
It work fine if we reload page.

Tensorflow uses Promise to download model and there is no way to cancel promise. I tried hack using `promise.race([debouncePromise, downloadModelPromise]) `with debounce of 10 sec of progress difference. But even if `debouncePromise` calls first `downloadModelPromise` keep running and showing progress. So technically it won't be possible to cancel promise.

I tried with restart download process but it reiterate till 92% and again stuck. Seems like tf.loadGraphModel() internally check if model downloading or not. 

**Just would like to know that, Can it be possible to reinitiate download model process?** :)
